src/ensembl/src/root/Root.test.tsx

```diff
@@ -1,29 +1,62 @@
 import React from 'react';
-import { shallow } from 'enzyme';
-import toJson from 'enzyme-to-json';
+import { mount } from 'enzyme';
 
-import Root from '../Root';
+import { Root } from './Root';
 import Header from '../header/Header';
 import Content from '../content/Content';
 
+jest.mock('../header/Header', () => () => 'Header');
+jest.mock('../content/Content', () => () => 'Content');
+jest.mock('../shared/privacy-banner/PrivacyBanner', () => () => (
+  <div className="privacyBanner">PrivacyBanner</div>
+));
+
+const cookiesMock: any = {
+  get: jest.fn()
+};
+const updateBreakpointWidth = jest.fn();
+
 describe('<Root />', () => {
   let wrapper: any;
 
+  const defaultProps = {
+    cookies: cookiesMock,
+    breakpointWidth: 0,
+    updateBreakpointWidth: updateBreakpointWidth
+  };
+  const getRenderedRoot = (props: any) => <Root {...props} />;
+
   beforeEach(() => {
-    wrapper = shallow(<Root />);
+    wrapper = mount(getRenderedRoot(defaultProps));
   });
 
-  describe('contains', () => {
-    test('<Header />', () => {
-      expect(wrapper.contains(<Header />)).toBe(true);
-    });
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  test('contains Header', () => {
+    expect(wrapper.contains(<Header />)).toBe(true);
+  });
 
-    test('<Content />', () => {
-      expect(wrapper.contains(<Content />)).toBe(true);
-    });
+  test('contains Content', () => {
+    expect(wrapper.contains(<Content />)).toBe(true);
   });
 
-  test('renders correctly', () => {
-    expect(toJson(wrapper)).toMatchSnapshot();
+  test('calls updateBreakpointWidth on mount', () => {
+    expect(updateBreakpointWidth).toHaveBeenCalled();
+  });
+
+  test('shows privacy banner if privacy cookie is not set', () => {
+    cookiesMock.get = jest.fn(() => '');
+    const wrapper = mount(getRenderedRoot(defaultProps));
+    expect(wrapper.find('.privacyBanner').length).toBe(1);
+  });
+
+  test('does not show privacy banner if privacy cookie is set', () => {
+    cookiesMock.get = jest.fn(() => 'true');
+    const wrapper = mount(getRenderedRoot(defaultProps));
+
+    // useEffect will be run asynchronously; so simulate asynchrony here
+    setTimeout(() => expect(wrapper.find('.privacyBanner').length).toBe(0), 0);
```

Comment:
Is this workaround similar to the following?
`// ugly hack: fall back to the end of event queue, giving priority to useEffect and useState
	      await new Promise((resolve) => setTimeout(resolve, 0));`

Reply:
Yep :-)

Reply:
Cool! I was wondering if it's better to keep it consistent so that it's easier to find and update these once enzyme starts to support it.

Reply:
Updated for consistency.

_(Also, I have a nagging suspicion that this particular implementation was not even working properly. I suspect `jest` exited this test before the `expect` was called. But I didn't check to make sure that was the case.)_